x/staking/types/errors.go

```diff
@@ -11,45 +11,53 @@ import (
+	ErrInsufficientShares                        = sdkerrors.Register(ModuleName, 22, "insufficient delegation shares")
+	ErrDelegationValidatorEmpty                  = sdkerrors.Register(ModuleName, 23, "cannot delegate to an empty validator")
+	ErrNotEnoughDelegationShares                 = sdkerrors.Register(ModuleName, 24, "not enough delegation shares")
+	ErrNotMature                                 = sdkerrors.Register(ModuleName, 25, "entry not mature")
+	ErrNoUnbondingDelegation                     = sdkerrors.Register(ModuleName, 26, "no unbonding delegation found")
+	ErrMaxUnbondingDelegationEntries             = sdkerrors.Register(ModuleName, 27, "too many unbonding delegation entries for (delegator, validator) tuple")
+	ErrNoRedelegation                            = sdkerrors.Register(ModuleName, 28, "no redelegation found")
+	ErrSelfRedelegation                          = sdkerrors.Register(ModuleName, 29, "cannot redelegate to the same validator")
+	ErrTinyRedelegationAmount                    = sdkerrors.Register(ModuleName, 30, "too few tokens to redelegate (truncates to zero tokens)")
+	ErrBadRedelegationDst                        = sdkerrors.Register(ModuleName, 31, "redelegation destination validator not found")
+	ErrTransitiveRedelegation                    = sdkerrors.Register(ModuleName, 32, "redelegation to this validator already in progress; first redelegation to this validator must complete before next redelegation")
+	ErrMaxRedelegationEntries                    = sdkerrors.Register(ModuleName, 33, "too many redelegation entries for (delegator, src-validator, dst-validator) tuple")
+	ErrDelegatorShareExRateInvalid               = sdkerrors.Register(ModuleName, 34, "cannot delegate to validators with invalid (zero) ex-rate")
+	ErrBothShareMsgsGiven                        = sdkerrors.Register(ModuleName, 35, "both shares amount and shares percent provided")
+	ErrNeitherShareMsgsGiven                     = sdkerrors.Register(ModuleName, 36, "neither shares amount nor shares percent provided")
+	ErrInvalidHistoricalInfo                     = sdkerrors.Register(ModuleName, 37, "invalid historical info")
+	ErrNoHistoricalInfo                          = sdkerrors.Register(ModuleName, 38, "no historical info found")
+	ErrEmptyValidatorPubKey                      = sdkerrors.Register(ModuleName, 39, "empty validator public key")
+	ErrCommissionLTMinRate                       = sdkerrors.Register(ModuleName, 40, "commission cannot be less than min rate")
+	ErrUnbondingNotFound                         = sdkerrors.Register(ModuleName, 41, "unbonding operation not found")
+	ErrUnbondingOnHoldRefCountNegative           = sdkerrors.Register(ModuleName, 42, "cannot un-hold unbonding operation that is not on hold")
+	ErrNotEnoughBalance                          = sdkerrors.Register(ModuleName, 43, "not enough balance")
+	ErrTokenizeShareRecordNotExists              = sdkerrors.Register(ModuleName, 44, "tokenize share record not exists")
```

Comment:
nit: (probably not for this PR) but we should define some kind of standard in how we write errors. These 3 seem to be saying that whatever you were looking for is not found but they are written differently. Although we have code completion, so we have less to guess :P

```
ErrNoValidatorFound
ErrUnbondingNotFound
ErrTokenizeShareRecordNotExists
```